src/lua/functions/core/game/game_functions.cpp

```diff
@@ -411,18 +417,16 @@ int GameFunctions::luaGameCreateMonster(lua_State* L) {
 		g_callbacks().executeCallback(EventCallback_t::monsterOnSpawn, &EventCallback::monsterOnSpawn, monster, position);
 		auto mtype = monster->getMonsterType();
 		if (mtype && mtype->info.raceid > 0 && mtype->info.bosstiaryRace == BosstiaryRarity_t::RARITY_ARCHFOE) {
-			SpectatorHashSet spectators;
-			g_game().map.getSpectators(spectators, monster->getPosition(), true);
+			auto spectators = Spectators().find<Player>(monster->getPosition(), true);
 			for (Creature* spectator : spectators) {
```

Comment:
const auto?

Reply:
old get method return all creatures, with find you could use auto and remove: `const auto tmpPlayer = spectator->getPlayer()`, no?